components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java

```diff
@@ -255,12 +250,37 @@ public static ExtendedAPI generateAPIFromOpenAPIDef(OpenAPI openAPI, String open
         api.setId(apiId);
         api.setName(openAPI.getInfo().getTitle());
         api.setVersion(openAPI.getInfo().getVersion());
-        api.setTransport(Arrays.asList("http", "https"));
+        api.setTransport(getTransport(openAPI));
         //open API content should be set in json in order to validation filter to work.
         api.setApiDefinition(openAPIContent);
         return api;
     }
 
+    private static List<String> getTransport(OpenAPI openAPI) {
+        List<String> transports = new ArrayList<>();
+        Map<String, Object> apiDefExtensions = openAPI.getExtensions();
+        if (apiDefExtensions.containsKey(OpenAPIConstants.TRANSPORT_SECURITY)) {
+            logger.debug(OpenAPIConstants.TRANSPORT_SECURITY + " extension found in the API Definition");
+            try {
+                Object transportSecurityExtension = apiDefExtensions.get(OpenAPIConstants.TRANSPORT_SECURITY);
+                TransportSecurity transportSecurity =
```

Comment:
Can't we directly use the extension here rather than assigning it to a object and then passing it to objectmapper